rholang/src/main/scala/coop/rchain/rholang/interpreter/Interpreter.scala

```diff
@@ -25,36 +26,38 @@ object Interpreter {
 
   def apply[F[_]](implicit interpreter: Interpreter[F]): Interpreter[F] = interpreter
 
-  implicit def interpreter[F[_]](implicit F: Sync[F]): Interpreter[F] = new Interpreter[F] {
+  implicit def interpreter[F[_]](
+      implicit M: Monad[F],
+      S: Sync[F],
+      C: _cost[F],
+      E: _error[F]
+  ): Interpreter[F] =
+    new Interpreter[F] {
 
-    def evaluate(runtime: Runtime[F], term: String): F[EvaluateResult] =
-      evaluate(runtime, term, Cost.Max)
+      def evaluate(runtime: Runtime[F], term: String): F[EvaluateResult] =
+        evaluate(runtime, term, Cost.Max)
 
-    def evaluate(runtime: Runtime[F], term: String, initialPhlo: Cost): F[EvaluateResult] = {
-      implicit val rand: Blake2b512Random = Blake2b512Random(128)
-      for {
-        checkpoint <- runtime.space.createCheckpoint()
-        res        <- injAttempt(runtime.reducer, runtime.errorLog, term, initialPhlo)
-        _          <- if (res.errors.nonEmpty) runtime.space.reset(checkpoint.root) else F.unit
-      } yield res
-    }
+      def evaluate(runtime: Runtime[F], term: String, initialPhlo: Cost): F[EvaluateResult] = {
+        implicit val rand: Blake2b512Random = Blake2b512Random(128)
+        for {
+          checkpoint <- runtime.space.createCheckpoint()
+          res        <- injAttempt(runtime.reducer, runtime.errorLog, term, initialPhlo)
+          _          <- if (res.errors.nonEmpty) runtime.space.reset(checkpoint.root) else M.pure(())
+        } yield res
+      }
 
-    def injAttempt(
-        reducer: ChargingReducer[F],
-        errorLog: ErrorLog[F],
-        term: String,
-        initialPhlo: Cost
-    )(implicit rand: Blake2b512Random): F[EvaluateResult] = {
-      val parsingCost = accounting.parsingCost(term)
-      // TODO: charge(parsingCost) so that it is visible in the cost log
-      val phloAfterParsing = initialPhlo - parsingCost
-      if (phloAfterParsing.value <= 0)
-        EvaluateResult(parsingCost, Vector(OutOfPhlogistonsError)).pure[F]
-      else
+      def injAttempt(
+          reducer: ChargingReducer[F],
+          errorLog: ErrorLog[F],
+          term: String,
+          initialPhlo: Cost
+      )(implicit rand: Blake2b512Random): F[EvaluateResult] = {
+        val parsingCost = accounting.parsingCost(term)
         ParBuilder[F].buildNormalizedTerm(term).attempt.flatMap {
           case Right(parsed) =>
             for {
-              _         <- reducer.setPhlo(phloAfterParsing)
+              _         <- reducer.setPhlo(initialPhlo)
+              _         <- charge[F](parsingCost).attempt
```

Comment:
1) I need to attempt here, otherwise `charge` raises an error in `F`, but all callers rely on `EvaluateResult` being returned. This line will disappear anyway when it's incorporated into `buildNormalizedTerm` so I guess this shouldn't be a blocker for this PR.
2) Will do

Reply:
I'm going to add handling of the .attempt